web3/_utils/events.py

```diff
@@ -255,6 +256,10 @@ def get_event_data(
     normalized_log_data = map_abi_data(
         BASE_RETURN_NORMALIZERS, log_data_types, decoded_log_data
     )
+    named_log_data = named_tree(
```

Comment:
See my comment on `decode_function_input()`... I think it may be ideal to created `ABIDecodedNamedTuples` only if requested by the user. This probably looks like a flag on this method which, when called internally in `contract.py` and `async_contract.py` would be just passing along the `contract.decode_tuples` flag.